cedar-policy/src/frontend/validate.rs

```diff
@@ -18,136 +18,129 @@
+        Ok((policies, schema)) => {
+            let validator = Validator::new(schema);
+            let validation_result = validator.validate(&policies, ValidationMode::default());
+            let validation_errors: Vec<ValidationError> = validation_result
+                .validation_errors()
+                .map(|error| ValidationError {
+                    policy_id: error.location().policy_id().to_string(),
+                    error: format!("{}", error.error_kind()),
+                })
+                .collect();
+            let validation_warnings: Vec<ValidationWarning> = validation_result
+                .validation_warnings()
+                .map(|error| ValidationWarning {
+                    policy_id: error.location().policy_id().to_string(),
+                    warning: format!("{}", error.warning_kind()),
+                })
+                .collect();
+            ValidationAnswer::Success {
+                validation_errors,
+                validation_warnings,
             }
         }
+        Err(errors) => ValidationAnswer::Failure { errors },
     }
-
-    if !parse_errors.is_empty() {
-        return Ok(ValidateAnswer::ParseFailed {
-            errors: parse_errors,
-        });
-    }
-
-    let schema = call
-        .schema
-        .clone()
-        .try_into()
-        .map_err(|e| format!("could not construct schema: {e}"))?;
-    let validator = Validator::new(schema);
-
-    let notes: Vec<ValidationNote> = validator
-        .validate(
-            &policy_set,
-            cedar_policy_validator::ValidationMode::default(),
-        )
-        .validation_errors()
-        .map(|error| ValidationNote {
-            policy_id: error.location().policy_id().to_string(),
-            note: format!("{}", error.error_kind()),
-        })
-        .collect();
-
-    Ok(ValidateAnswer::Success { notes })
 }
 
 /// public string-based validation function
 pub fn json_validate(input: &str) -> InterfaceResult {
-    serde_json::from_str::<ValidateCall>(input).map_or_else(
+    serde_json::from_str::<ValidationCall>(input).map_or_else(
         |e| InterfaceResult::fail_internally(format!("error parsing call: {e:}")),
-        |call| match validate(&call) {
-            Ok(answer @ ValidateAnswer::Success { .. }) => InterfaceResult::succeed(answer),
-            Ok(ValidateAnswer::ParseFailed { errors }) => InterfaceResult::fail_bad_request(errors),
-            Err(e) => InterfaceResult::fail_internally(e),
+        |call| match validate(call) {
+            answer @ ValidationAnswer::Success { .. } => InterfaceResult::succeed(answer),
+            ValidationAnswer::Failure { errors } => InterfaceResult::fail_bad_request(errors),
         },
     )
 }
 
+/// Struct containing the input data for validation
 #[derive(Serialize, Deserialize)]
 #[cfg_attr(feature = "wasm", derive(tsify::Tsify))]
 #[cfg_attr(feature = "wasm", tsify(into_wasm_abi, from_wasm_abi))]
-struct ValidateCall {
```

Comment:
The `Validate` to `Validation` change looks nice, but I wonder if it's worth making a breaking change to the API. Depends on how much we care about keeping this interface stable.

Reply:
I _think_ this isn't breaking since the `ValidateCall` type was only ever exposed indirectly via its serialized JSON format.